UnitWorld_server/server/networking/ClientsGameSender.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include "PlayerClient.h"
+
+#include "shared/game/GameManager.h"
+
+#include "shared/transfers/PhysicsCommunicationAssembler.h"
+
+#include "shared/communication/MessageSerializer.h"
+#include "shared/communication/messages/CompleteGameStateMessage.h"
+
+#include "shared/communication/messages/game/CommunicatedPlayer.h"
+#include "shared/communication/messages/game/CommunicatedSinguity.h"
+
+#include "communications/CommunicationHandler.h"
+
+#include <chrono>
+#include <mutex>
+
+namespace uw
+{
+    class ClientsGameSender
+    {
+    public:
+        ClientsGameSender(std::shared_ptr<GameManager> gameManager, std::shared_ptr<MessageSerializer> messageSerializer, std::shared_ptr<PhysicsCommunicationAssembler> physicsCommunicationAssembler) :
+            _gameManager(gameManager),
+            _messageSerializer(messageSerializer),
+            _maxMsBetweenCompleteStates(60000 / MINIMAL_COMPLETE_STATE_PER_MINUTE),
+            _isRunning(true),
+            _lastSendCompleteState(std::chrono::steady_clock::now()),
+            _physicsCommunicationAssembler(physicsCommunicationAssembler)
+        {}
+
+        void addClient(const PlayerClient& playerClient)
+        {
+            _playerClients = _playerClients.push_back(playerClient);
```

Comment:
Could `_playerClients` be passed as a *constructor* argument instead?

Reply:
This would imply all players of a game must be created before the game itself. Is that a thing we can do?

Reply:
Adding a player while a game is running feels odd to me, that's why

Reply:
Yes we could, but I want to be able to join an AI game after the start as an observer, but implementing the two types of add was more work than only doing this one.

I'll keep it like this for now.